src/hooks/useSchedule.tsx

```diff
@@ -0,0 +1,120 @@
+import { useState, useEffect, useCallback, useMemo } from 'react';
+import dayjs from 'dayjs';
+import isSameOrAfter from 'dayjs/plugin/isSameOrAfter';
+import isSameOrBefore from 'dayjs/plugin/isSameOrBefore';
+
+dayjs.extend(isSameOrBefore);
+dayjs.extend(isSameOrAfter);
+
+export const useSchedule = (startDateStr: string, endDateStr: string) => {
+  const MAX_DATE = 4;
+  const TOTAL_TIME = 16;
```

Comment:
요 상수들은 useSchedule 바깥으로 빼어도 좋을 것 같네용
```suggestion
const MAX_DATE = 4;
const TOTAL_TIME = 16;

export const useSchedule = (startDateStr: string, endDateStr: string) => {
  const startDate = ...
```

Reply:
해당 상수들이 useCallback 에서 사용되어서 우선 hook 내부에 뒀다가 리팩토링할때 빼도 괜찮을까욥?? 

지금 웹 환경에서는 2차원 스크롤이 잘되는데 모바일 환경에서는 세로로만 스크롤이 가능해요ㅠ,ㅠ (2차원 스크롤이 안되고있습니다..) 혹시 터치이벤트 관련해서 도움 받을 수 있다면 가로로 왜 인식이 안되는지 궁금합니다..

Reply:
> 해당 상수들이 useCallback 에서 사용되어서 우선 hook 내부에 뒀다가 리팩토링할때 빼도 괜찮을까욥??

네 그럼용! 그냥 단순 궁금증인데, hook 외부로 빼내면 에러가 발생하는건가요?? 그렇다면 무슨 에러인지 공유해주실 수 있을까용

Reply:
> 지금 웹 환경에서는 2차원 스크롤이 잘되는데 모바일 환경에서는 세로로만 스크롤이 가능해요ㅠ,ㅠ (2차원 스크롤이 안되고있습니다..) 혹시 터치이벤트 관련해서 도움 받을 수 있다면 가로로 왜 인식이 안되는지 궁금합니다..

혹시 첨부해주신 동영상은 어느 환경에서 실행한 것인가요? 스토리북에서는 동작하지 않는 것 같아 여쭈어보아용

Reply:
`React Hook useCallback has a missing dependency: 'savedTimeSlots'. Either include it or remove the dependency array. Outer scope values like 'TOTAL_TIME' aren't valid dependencies because mutating them doesn't re-render the component ` 다음과 같은 에러가 납니당


지금은 `  // eslint-disable-next-line react-hooks/exhaustive-deps` 요 주석으로 lint warining 막아놨어욥

Reply:
엇 저도 웹/모바일 둘다 스토리북 들어가서 테스트했어요!

Reply:
> React Hook useCallback has a missing dependency: 'savedTimeSlots'. Either include it or remove the dependency array. Outer scope values like 'TOTAL_TIME' aren't valid dependencies because mutating them doesn't re-render the component  다음과 같은 에러가 납니당
지금은   // eslint-disable-next-line react-hooks/exhaustive-deps 요 주석으로 lint warining 막아놨어욥

아항 브랜치 체크아웃해서 확인해봤는데, 훅 바깥으로 꺼내면 렌더링과 관계없이 유지(새로 생성되지 않으니)되니까 디펜던시 어레이에서 TOTAL_TIME을 제거해 주면 warning이 발생하지 않을 것 같네요!

상수값과 별개로 savedTimeSlots은 메모처리해 두셨으니 lint 막는 주석 두신거 이해했습니다 👍

Reply:
엇 저는 잘돼요.. 이럼 안되는데

Reply:
<img width="312" alt="스크린샷 2024-09-16 오후 9 14 24" src="https://github.com/user-attachments/assets/de9eb4ab-6380-4170-a572-88159ae9f84a">

요 상태가 스토리북에 작성된 TimeRangePicker 기본값이 맞나요?! 저는 여기서 아무 클릭도 먹히지 않는 중이에요....😂 사파리랑 크롬 모두 동일해용...